src/Visualization/Python/PlotDatFile.py

```diff
@@ -0,0 +1,296 @@
+    """Opens the specified HDF5 file.
+
+    If no subfile is specified this function prints all available Dat
+    subfiles in the root of the HDF5 and then exists. If a subfile is
+    specified, the function forwards on to process_subfile()
+    """
+    def available_subfiles(h5file):
+        return [member for member in h5file.keys() if member.endswith('.dat')]
+
+    with h5py.File(filename) as h5file:
+        if subfile_name != None:
+            dat_file = h5file.get(subfile_name + '.dat')
+            if dat_file is None:
+                raise Exception(
+                    "Unable to open dat file '%s'. Available files are:\n%s" %
+                    (subfile_name + '.dat', available_subfiles(h5file)))
+            return process_subfile(dat_file, **kwargs)
+        else:
+            print("Available dat files are:", available_subfiles(h5file))
+            return available_subfiles(h5file)
+    return None
+
+
+def parse_args():
+    """
+    Parse the command line arguments
+    """
+    import argparse as ap
+    import json
+
+    # We use a custom formatter to allow us to override the default
+    # formatting style when we want to add line breaks ourselves.
+    class SmartFormatter(ap.HelpFormatter):
+        def _split_lines(self, text, width):
+            if text.startswith('R|'):
+                return text[2:].splitlines()
+            # If the user did not ask for raw formatting use argparse's
+            # formatter instead.
+            return ap.HelpFormatter._split_lines(self, text, width)
+
+    parser = ap.ArgumentParser(
+        description=
+        "Analyze reduction data files (.dat HDF5 subfiles) written out "
+        "by SpECTRE. Plots of the data can be generated, looking at "
+        "what variables were written to the subfile, and writing the "
+        "dat subfile to a text file on disk.",
+        formatter_class=SmartFormatter)
+    parser.add_argument('filename', help="The HDF5 file to read.")
+    parser.add_argument(
+        '--subfile-name',
+        '-d',
+        required=False,
+        help="The dat subfile to read excluding the .dat extension. "
+        "If excluded all available dat subfiles will be printed.")
+
+    action_group = parser.add_mutually_exclusive_group(required=True)
+
+    action_group.add_argument('--legend-only',
+                              action='store_true',
+                              help="If specified, only print out the legend")
+    action_group.add_argument(
+        '--write-dat',
+        help="If specified, writes the entire dat file to a text Dat file with "
+        "spaces as the delimiter and exits.")
+
+    action_group.add_argument(
+        '--output',
+        '-o',
+        help="Name of the plot file. The '.pdf' extension will "
+        "be added automatically. Note: Requires additional arguments to be "
+        "specified. Specify --help alongside '--output' or '-o' to see "
+        "additional arguments.")
```

Comment:
I didn't catch this before, but when I do
```
python PlotDatFile.py --help -o
```

I get the expected help message. However, if I do

```
python PlotDatFile.py -o --help
```

I get this error message:

```
PlotDatFile.py: error: argument --output/-o: expected one argument
```

Is there a way to parse this so that it looks for `--help` first before it parses `-o` so we avoid this error?

Reply:
I have no control over the parser. This just uses python's standard parser